applications/mlflow/charts/mlflow/templates/postgres-backup-deployment.yaml

```diff
@@ -0,0 +1,65 @@
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  name: {{ include "mlflow.fullname" . }}-postgres-backup
+spec:
+  replicas: 1
+  selector:
+    matchLabels:
+      app: db-backup-restore
+  template:
+    metadata:
+      labels:
+        app: db-backup-restore
+      annotations:
+        backup.velero.io/backup-volumes: backup
+        pre.hook.backup.velero.io/command: '["/scripts/db-backup.sh"]'
+        pre.hook.backup.velero.io/timeout: "10m"
+    spec:
+      volumes:
+        - name: backup
+          emptyDir: {}
+        - name: scripts
+          configMap:
+            name: {{ include "mlflow.fullname" . }}-backup-scripts
+            defaultMode: 0777
+      initContainers:
+        - name: wait-for-postgresql
+          image: docker.io/bitnami/postgresql:15.3.0-debian-11-r0
+          imagePullPolicy: {{ .Values.mlflow.image.pullPolicy }}
+          command:
+            [
+              "sh",
+              "-c",
+              "until PGPASSWORD=$POSTGRES_PASSWORD psql -U $POSTGRES_USER -h $POSTGRES_HOST -p 5432 -d $POSTGRES_DB -c 'SELECT 1'; do sleep 1; done;",
+            ]
+          envFrom:
+            - secretRef:
+                name: {{ printf "%s-waitfor-postgres" (include "mlflow.fullname" .)  | trunc 63 | trimAll "-" }}
+        - name: restore-db
+          image: docker.io/bitnami/postgresql:15.3.0-debian-11-r0
```

Comment:
Why an initContainer that always does a restore? The restore should be done by the hooks as well right?

Reply:
As mentioned earlier, the initContainer adopts this pattern https://github.com/replicatedhq/kots/blob/710faf95e583827812a4ac41245bbf2e14b33af9/pkg/kotsadm/objects/kotsadm_objects.go#L873-L986, and we will remove the file after the restore, so it is ensure to run only once.